server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorDVLeafFieldData.java

```diff
@@ -57,10 +62,16 @@ public DocValuesScriptFieldFactory getScriptFieldFactory(String name) {
                     // values that can be iterated through. Since VectorValues.EMPTY throws on docID(), pass a null instead.
                     values = null;
                 }
-                return new KnnDenseVectorDocValuesField(values, name, dims);
+                return switch (elementType) {
+                    case BYTE -> new ByteKnnDenseVectorDocValuesField(values, name, elementType, dims);
+                    case FLOAT -> new KnnDenseVectorDocValuesField(values, name, elementType, dims);
+                };
             } else {
                 BinaryDocValues values = DocValues.getBinary(reader, field);
-                return new BinaryDenseVectorDocValuesField(values, name, dims, indexVersion);
+                return switch (elementType) {
+                    case BYTE -> new ByteBinaryDenseVectorDocValuesField(values, name, elementType, dims);
+                    case FLOAT -> new BinaryDenseVectorDocValuesField(values, name, elementType, dims, indexVersion);
+                };
```

Comment:
SO CLEAN! Dang, I love these switch expressions. Thank you Java for slowly modernizing!

Reply:
This is MUCH nicer.